src/main/java/be/acara/frontend/controller/UserController.java

```diff
@@ -59,21 +59,37 @@ public String registration(@Valid @ModelAttribute(ATTRIBUTE_USER_FORM) UserModel
     
     @GetMapping("/detail/{id}")
     public String displayUser(@PathVariable("id") Long id, ModelMap model,
-                              @RequestParam(name = "page", defaultValue = "1", required = false) int page,
-                              @RequestParam(name = "size", defaultValue = "20", required = false) int size) {
+                              @RequestParam(name = "pageSubscribedEvents", defaultValue = "1", required = false) int pageSubscribedEvents,
+                              @RequestParam(name = "sizeSubscribedEvents", defaultValue = "3", required = false) int sizeSubscribedEvents,
+                              @RequestParam(name = "pageLikedEvents", defaultValue = "1", required = false) int pageLikedEvents,
+                              @RequestParam(name = "sizeLikedEvents", defaultValue = "3", required = false) int sizeLikedEvents) {
         User user = userService.getUser(id);
-        EventDtoList events = eventService.getEventsFromUser(id, page-1, size);
-        int totalPages = events.getTotalPages();
-        if (totalPages > 0) {
-            List<Integer> pageNumbers = IntStream.rangeClosed(1, totalPages)
-                    .boxed()
-                    .collect(Collectors.toList());
-            model.addAttribute("pageNumbers", pageNumbers);
+        EventDtoList subscribedEvents = eventService.getEventsFromUser(id, pageSubscribedEvents-1, sizeSubscribedEvents);
+        List<Integer> pageNumbersSubscribedEvent = getPageNumbers(subscribedEvents);
+        if(pageNumbersSubscribedEvent != null) {
+            model.addAttribute("pageNumbersSubscribedEvents", pageNumbersSubscribedEvent);
+        }
+        EventDtoList likedEvents = eventService.getEventsThatUserLiked(id, pageLikedEvents-1, sizeLikedEvents);
+        List<Integer> pageNumbersLikedEvents = getPageNumbers(likedEvents);
+        if(pageNumbersLikedEvents != null) {
+            model.addAttribute("pageNumbersLikedEvents", pageNumbersLikedEvents);
         }
         model.addAttribute(ATTRIBUTE_USER, user);
-        model.addAttribute(ATTRIBUTE_EVENTS, events);
+        model.addAttribute("subscribedEvents", subscribedEvents);
+        model.addAttribute("likedEvents", likedEvents);
         return "user/userDetails";
     }
+
+    private List<Integer> getPageNumbers(EventDtoList events) {
+        List<Integer> pageNumbers = null;
+        int totalPagesLikedEvents = events.getTotalPages();
+        if (totalPagesLikedEvents > 0) {
+            pageNumbers = IntStream.rangeClosed(1, totalPagesLikedEvents)
+                    .boxed()
+                    .collect(Collectors.toList());
+        }
+        return pageNumbers;
+    }
```

Comment:
```
private void addPageNumbers(EventDtoList events, ModelMap modelMap, String attribute) {
        if (events.getTotalPages() == 0) {
            return;
        }
        modelMap.addAttribute(attribute, IntStream.rangeClosed(1, events.getTotalPages())
                .boxed()
                .collect(Collectors.toList()));
    }
```

Reply:
We could put this in an util method? its almost generic enough to pull it off. 🤔